mslice/plotting/plot_window/slice_plot.py

```diff
@@ -342,19 +342,22 @@ def _run_temp_dependent(self, slice_plotter_method, previous):
                 self._slice_plotter_presenter.add_sample_temperature_field(temp_value)
                 self._slice_plotter_presenter.update_sample_temperature(self.ws_name)
             else:
-                try:
-                    temp_value = float(temp_value)
-                    if temp_value < 0:
-                        raise ValueError
-                except ValueError:
-                    self.plot_window.error_box("Invalid value entered for sample temperature. Enter a value in Kelvin \
+                if temp_value is not None and temp_value.strip():
+                    if temp_value.endswith('K'):
+                        temp_value = temp_value[:-1]
+                    try:
+                        temp_value = float(temp_value)
+                        if temp_value < 0:
+                            raise ValueError
+                    except ValueError:
+                        self.plot_window.display_error("Invalid value entered for sample temperature. Enter a value in Kelvin \
                                                or a sample log field.")
```

Comment:
Perhaps, we could use https://github.com/mantidproject/mslice/blob/7c5a6545548e54a9f965b841f30c63365a3f5157/mslice/models/slice/slice_functions.py#L145-L154
to get temperature 
``temp_value = get_sample_temperature_from_string(temp_value)``  
that leads to ``temp_value`` being either ``None`` or `` Float``. 

Then, we can check if there is an error and raise if needed
``if raise_temperature_error(temp_value):``
      ``self.plot_window.display_error("Invalid value entered for sample temperature. \``
                ``Enter a value in Kelvin or a sample log field.")``

where raise_temperature_error defined something like below

`def raise_temperature_error(temp_value):`
        `raise_error = (temp_value is None`)
        `if raise_error:`
               `pass`
         `else:`
               `raise_error = (temp_value < 0)`
 `return raise_error`